Assets/_Laz/Scripts/Lazo/Lazo.cs

```diff
@@ -39,29 +40,15 @@ private float TravelledDistance
         public event Action<float> OnLazoLimitChanged;
         public event Action OnLazoDeactivated;
 
-        public bool IsLazoing
-        {
-            get => _isLazoing;
-            set
-            {
-                _isLazoing = value;
-                if (!_isLazoing)
-                {
-                    Reset();
-                    if (OnLazoDeactivated != null)
-                    {
-                        OnLazoDeactivated();
-                    }
-                    // Debug
-                    DebugClearListOfCubes();
-                }
-            }
-        }
+        public bool IsLazoing => _isLazoing;
```

Comment:
Same here, put the logic into a method instead of a setter function, so it's more understandable